File: No_2350/2350. Shortest Impossible Sequence of Rolls.py
class Solution:
    def shortestSequence(self, rolls: List[int], k: int) -> int:
        count = 0
        hset = set()
        
        for num in rolls:
            hset.add(num)
            if len(hset) == k:
                hset.clear()
                count += 1
                
        return count + 1

File: No_2350/2350. Shortest Impossible Sequence of Rolls.cpp
class Solution {
public:
    int shortestSequence(vector<int>& rolls, int k) {
        int count = 0;
        unordered_set<int> hset;
        for (int num : rolls) {
            hset.insert(num);
            if (hset.size() == k) {
                hset.clear();
                count++;
            }
        }
        
        return count + 1;
    }
};